### DISCUSSION TITLE
Can V be used for arduino?

### DISCUSSION THREAD

**Discussion by skyleriscoding** [vlang/v#17467] [Questions and Answers]
I'm currently working on an arduino project for my lyceum's computer science course. I'm curious if V, a language that can be compiled to C and is advertised to work with embedded systems, can be used to develop arduino programs.

**Comment by rraammiinn** [comment]
it's a good idea .
personally I both don't know c++ and don't like it .
since v uses c instead of llvm it should be possible .

**Comment by medvednikov** [comment]
Yes, it can be used anywhere C is used.

**Comment by enginux** [comment]
Firstly, search How to code Arduino in C.
Secondly, v2c.

 ATMega MCUs can be program with embedded C.

**Comment by AntioneMcguyer9dS** [comment]
Yes, V is a programming language that can be compiled to C, and it is designed to work with embedded systems. It should be possible to use V to develop Arduino programs, although it may require some additional configuration or setup.

One thing to keep in mind is that the Arduino platform has its own specific libraries and functions that are used to interact with the hardware. These libraries are typically written in C or C++, and it may be necessary to use them in conjunction with V in order to develop Arduino programs.

However, V's support for C interoperability should make it possible to use these libraries from within V code. Additionally, V's focus on performance and low-level programming may make it a good fit for developing embedded systems, like those used in Arduino projects.

If you are interested in using V for your Arduino project, it is recommended that you do some additional research and experimentation to determine if it is a good fit for your specific needs.

**Comment by fermarsan** [comment]
There is a beta project named "Aixt" that implements a subset of V syntax for microcontrollers, including some "Arduinos".

https://github.com/fermarsan/aixt
